src/plugins/chart_expressions/expression_heatmap/public/compoments/heatmap_component.tsx

```diff
@@ -349,56 +432,65 @@ export const HeatmapComponent: FC<HeatmapRenderProps> = ({
     timeZone,
   };
 
-  if (!chartData || !chartData.length) {
-    return <EmptyPlaceholder icon={LensIconChartHeatmap} />;
-  }
-
   return (
-    <Chart>
-      <Settings
-        onElementClick={onElementClick}
-        showLegend={args.legend.isVisible}
-        legendPosition={args.legend.position}
-        debugState={window._echDebugStateFlag ?? false}
-        theme={{
-          ...chartTheme,
-          legend: {
-            labelOptions: { maxLines: args.legend.shouldTruncate ? args.legend?.maxLines ?? 1 : 0 },
-          },
-        }}
-        xDomain={{
-          min: data.dateRange?.fromDate.getTime() ?? NaN,
-          max: data.dateRange?.toDate.getTime() ?? NaN,
-        }}
-        onBrushEnd={onBrushEnd as BrushEndListener}
-      />
-      <Heatmap
-        id={tableId}
-        name={valueColumn.name}
-        colorScale={{
-          type: 'bands',
-          bands,
-        }}
-        data={chartData}
-        xAccessor={args.xAccessor}
-        yAccessor={args.yAccessor || 'unifiedY'}
-        valueAccessor={args.valueAccessor}
-        valueFormatter={(v: number) => valueFormatter.convert(v)}
-        xScale={xScale}
-        ySortPredicate="dataIndex"
-        config={config}
-        xSortPredicate="dataIndex"
-      />
-    </Chart>
+    <>
+      {showLegend !== undefined && (
+        <LegendToggle
+          onClick={toggleLegend}
+          showLegend={showLegend}
+          legendPosition={args.legend.position}
+        />
+      )}
+      <Chart>
+        <Settings
+          onElementClick={onElementClick}
+          showLegend={showLegend ?? args.legend.isVisible}
+          legendPosition={args.legend.position}
+          legendColorPicker={uiState ? legendColorPicker : undefined}
+          debugState={window._echDebugStateFlag ?? false}
+          tooltip={tooltip}
+          theme={{
+            ...chartTheme,
+            legend: {
+              labelOptions: {
+                maxLines: args.legend.shouldTruncate ? args.legend?.maxLines ?? 1 : 0,
+              },
+            },
+          }}
+          xDomain={{
+            min:
+              dateHistogramMeta && dateHistogramMeta.timeRange
+                ? new Date(dateHistogramMeta.timeRange.from).getTime()
+                : NaN,
+            max:
+              dateHistogramMeta && dateHistogramMeta.timeRange
+                ? new Date(dateHistogramMeta.timeRange.to).getTime()
+                : NaN,
+          }}
+          onBrushEnd={onBrushEnd as BrushEndListener}
+        />
+        <Heatmap
+          id="heatmap"
```

Comment:
why removing the `tableId` here?
Can we do `id={tableId ?? 'heatmap'}`?

Reply:
We have only one table now (we dont use the multitable so we cant do this!

Reply:
I was looking at the `id` type description and it mentions that it should be unique:
```ts
export interface Spec {
    /** unique Spec identifier */
    id: string;
    /** Chart type define the type of chart that use this spec */
    chartType: ChartType;
    /** The type of spec, can be series, axis, annotation, settings etc */
    specType: string;
}
```

I've tested in dashboard and see no problem- I was assuming it was assigning the `id` to the canvas element. Maybe somebody from @elastic/datavis team can clarify the uniqueness constraint in this case?

Reply:
I discussed it offline with MarcoV and made it unique!